test/cql-pytest/test_service_levels.py

```diff
@@ -27,6 +28,42 @@ def new_service_level(cql):
         cql.execute(f"DETACH SERVICE LEVEL FROM {cql.cluster.auth_provider.username}")
         cql.execute(f"DROP SERVICE LEVEL IF EXISTS {sl}")
 
+@contextmanager
+def new_service_level_with_params(cql, timeout=None, workload_type=None):
+    params = ""
+    if timeout != None and workload_type != None:
+        params = f"WITH timeout = {timeout} AND workload_type = '{workload_type}'"
+    elif timeout != None:
+        params = f"WITH timeout = {timeout}"
+    elif workload_type != None:
+        params = f"WITH workload_type = '{workload_type}'"
+
+    try:
+        sl = f"sl_{unique_name()}"
+        cql.execute(f"CREATE SERVICE LEVEL {sl} {params}")
+        yield sl
+    finally:
+        cql.execute(f"DROP SERVICE LEVEL IF EXISTS {sl}")
```

Comment:
We should avoid proliferation of convenience functions and inventing new APIs. Those just make tests harder to read and harder to expand when in the future service levels get new parameters. Instead of the two new_service_level functions, you you can have just one function, new_service_level(), that takes an optional "params string", e.g., "WITH timeout=... workload_type=..." and uses that instead of building it.

Moreover, it's strange that one of these functions have attach/detach and the other doesn't and the caller does this. I think you should pass an optional "role" parameter,  and the single function can attach to the give role or the default one if missing.

Reply:
Merged that into one function